### PR TITLE
WIP: Emptydir and security context

### PR DESCRIPTION
**Pull request by pmorie**
WIP for #7925

Very rough, mostly to track the downstream work I'm doing in openshift

### CODE REVIEW

File: pkg/volume/empty_dir/empty_dir.go [hunk 1]
```diff
@@ -139,6 +147,7 @@ const (
 // These do not persist beyond the lifetime of a pod.
 type emptyDir struct {
 	podUID        types.UID
+	pod           *api.Pod
```
Comment (thockin) [inline pkg/volume/empty_dir/empty_dir.go]: do you ned PodUID then?

File: pkg/volume/empty_dir/empty_dir.go [hunk 2]
```diff
@@ -21,13 +21,20 @@ import (
...
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	volumeutil "github.com/GoogleCloudPlatform/kubernetes/pkg/volume/util"
 	"github.com/golang/glog"
 )
 
+// TODO: in the near future, this will be changed to be more restrictive
+// and the group will be set to allow containers to use emptyDir volumes
+// from the group attribute.
+const perm os.FileMode = 0777
```
Comment (thockin) [inline pkg/volume/empty_dir/empty_dir.go]: better name pls

Comment (pmorie) [inline pkg/volume/empty_dir/empty_dir.go]: @thockin a lot of this code is already present in #9384 just FYI, don't want you to duplicate review effort.

File: pkg/volume/empty_dir/empty_dir.go (hunk 1 above)
Comment (pmorie) [inline pkg/volume/empty_dir/empty_dir.go]: I don't, and I've taken it out of the downstream patch for this.  I've been pretty much embroiled in prep for the OpenShift launch but once the dust settles I will update this PR.

**Comment by pmorie** [comment]
I also expect this to fail until the `mounttest-user` image is pushed to gcr.io

**Comment by pmorie** [comment]
@thockin This is the next one I'm going to bug you about -- I'm going to refactor this code but I'm carrying a patch for this in OpenShift and so it makes sense for me to begin refactoring from this state.

**Comment by mrry550** [comment]
@pmorie - this one could use a little love too... likely goes with the other work?

**Comment by pmorie** [comment]
@mrry550 This PR is likely not going in.  #12944 supercedes it.

**Comment by mrry550** [comment]
Thx @pmorie - I'll watch over there instead. ;)

**Comment by pmorie** [comment]
Superceded by #14192, closing out.

**Pull request closed without merging**